### PR TITLE
chore: automatically merge dependabot PRs

### PR DESCRIPTION
**Pull request by gruyaume**
# Description

Automatically merge dependabot PRs.

# Checklist:

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

**Comment by gruyaume** [comment]
@saltiyazan anything else outside of the workflow I need to do here to get this working?

**Review by Gmerold — COMMENTED** [review]
Should we put that together with the rest of our workflows?

**Review by markbeierl — APPROVED** [review]

File: .github/workflows/dependabot_pr.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: "Dependabot Auto Approve and Merge"
+
+on:
+  pull_request:
+
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
```
Comment (Gmerold) [inline .github/workflows/dependabot_pr.yaml:10]: Should we put that together with the rest of our workflows?

**Comment by saltiyazan** [comment]
> @saltiyazan anything else outside of the workflow I need to do here to get this working?

Yes, you need to validate the branch protection rules, same thing you did in the TLS repos.
And you need to allow auto merge, but looking at the settings I see it is the case.
That should be it.

File: .github/workflows/dependabot_pr.yaml (hunk 1 above)
Comment (gruyaume) [reply .github/workflows/dependabot_pr.yaml:10]: Ok updated this to use the workflow added in this PR:
- https://github.com/canonical/sdcore-github-workflows/pull/35

File: .github/workflows/dependabot_pr.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  auto-merge:
+    uses: canonical/sdcore-github-workflows/.github/workflows/dependabot_pr.yaml@main
```
Comment (Gmerold) [inline .github/workflows/dependabot_pr.yaml:12]:
```suggestion
    uses: canonical/sdcore-github-workflows/.github/workflows/dependabot_pr.yaml@main
    secrets: inherit
```
The remote workflow uses `GITHUB_TOKEN`, which needs to be inherited from the calling job.

**Review by Gmerold — CHANGES_REQUESTED** [review]

File: .github/workflows/dependabot_pr.yaml (hunk 2 above)
Comment (gruyaume) [reply .github/workflows/dependabot_pr.yaml:12]: Does it need this? For example, the publish-charm workflow also uses the `GITHUB_TOKEN` secret, and it does not get inherited explicitly. I think there's something different about those default GitHub secrets.

Comment (Gmerold) [reply .github/workflows/dependabot_pr.yaml:12]: Yeah, looks like you're right. Sorry for confusion.

**Review by Gmerold — APPROVED** [review]

**Pull request merged**